src/GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp/Services/EventGridEventHandler.cs

```diff
@@ -16,10 +16,17 @@ public async Task<TResponse> Handle<TPayload, TResponse>(
         var payload = eventGridEvent.Data.ToObjectFromJson<TPayload>() 
                       ?? throw new Exception(
                           $"Unable to deserialise the payload of event into type {typeof(TPayload).Name}");
-        
-        var response = await handler(payload, context.CancellationToken);
 
-        logger.LogInformation("{FunctionName} completed. {Response}", context.FunctionDefinition.Name, response);
-        return response;
+        try
+        {
+            var response = await handler(payload, context.CancellationToken);
+            logger.LogInformation("{FunctionName} completed. {Response}", context.FunctionDefinition.Name, response);
+            return response;
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, "{FunctionName} errored processing {Request}.", context.FunctionDefinition.Name, eventGridEvent);
```

Comment:
Just wondering if you really intended this to be `{Request}` rather than `{Event}`?